ogticrd/covid-safe-paths
# Feat: Epidemiological discharge

#### Description:


Changing the previous implementation of epidemiological discharge, and use the new endpoint provided, also create the label
to users who are covid-19 free.

When you are testing thisPR take this into account:
- Review with multiple registered users in the app.
- The person who registers as mySelf is the only one who can activate or deactivate location sharing.
- When the person registered as mySelf becomes negative, the location sharing option must be deactivated if it is active.
- Anyone who was positive and turned negative should be removed from the app's positive support list.
- In the message that notifies you of their status, the nickname in which they registered should appear.
#### Linked issues:


Feat: [286](https://trello.com/c/iutqViaK/286-as-a-dev-i-want-to-replace-the-current-implementation-to-notify-users-when-they-change-epidemiological-status)
#### Screenshots:


![image](https://user-images.githubusercontent.com/43874147/103821171-095cb180-5044-11eb-8bc9-a64f0f5e2231.png)
![image](https://user-images.githubusercontent.com/43874147/103821180-0f529280-5044-11eb-8c14-8305f970da1f.png)
![image](https://user-images.githubusercontent.com/43874147/103821205-1aa5be00-5044-11eb-9cb0-90270ab36f8a.png)
![image](https://user-images.githubusercontent.com/43874147/103821216-1ed1db80-5044-11eb-9cef-47c9c18b2605.png)

#### How to test:


- `git pull && git checkout Feat/286-epidemiological-discharge`
- Register yourself by going to the register user section 
- Change your user state
- Close and Open the app again
- You should see a pop up notifying you the change
- Should move you out dependent of the change you made 

***Note:***

To test the PR you will have to modify its state manually, since now we do not have control of where the data is stored (You can do this by going to the *HomeScreen* index and on line 139 before calling the **validateUserState** function modify the userList changing the positive property for example: **userList [0] .positive = true** you can console log userList to see what data they have and what data they want to change)
